enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+```
+* `duration` of the command represnts the overall time required to complete the command. The label values will be bounded in the installer client to preset buckets of a certain step-size, 2.5 in the example above.
+* `return` - denotes success or failure. It could potentially be subsumed in result.
+* `result` label denotes a command outcome:
+    - `ProvisioningFailed` - for `create` the installer failed to provision resources
+    - `BootstrapFailed` - for `create` the bootstrap node failed
+    - `APIFailed` - for `create` and `wait-for` the control plane API never came up
+    - `OperatorsDegraded` - for `create` and `wait-for` command not all operators became available
+    - `Success` - for `create` and `wait-for` the cluster completed successfully and other commands completed successfully
+    - TODO: determine relevant results for other commands, perhaps inability to connect to SSH with `destroy` or `gather`
+
+#### cluster_installation_duration
+`cluster_installation_duration` tracks the duration of each stage of installation in the execution of the `create` and `wait-for` commands. Similar to `duration` in the previous stage, the value would be bound to certain buckets with a step size:
+
+```
+# HELP cluster_installation_duration keeps track of the count of the duration of individual stages of the command execution in buckets with the result of the command.
+# TYPE cluster_installation_duration counter
+cluster_installation_duration{infrastructure="5", bootstrap="5", api="5", operators="5" success="true"} 1
+```
+All label values would be rounded up to the nearest step size, so if it took 4 minutes to complete bootstrapping and the agreed upon step size was 2.5 minutes, the value would be rounded to 5.
+
+The proposed periods to track are:
+* `infrastructure`
+* `bootstrap`
+* `api`
+* `operators`
+
+One sample would be pushed per command invocation.
+
+#### cluster_installation_modification
+The `cluster_installation_modification` metric represents whether a user modified an installer created asset prior to running the `create` command.
+
+Feedback would be appreciated on two potential designs. One which aggregates the values into a single sample and one that pushes multiple samples:
+
+```
+# HELP cluster_installation_modification lists all the assets that were changed from the original value by the user before the create command was called.
+# TYPE cluster_installation_modification counter
+cluster_installation_modification{asset="cluster-config.yaml,cluster-scheduler-02-config.yml"} 1
+```
+In this case, the values would be sorted in order to reduce cardinality. The benefit of this approach is that it creates a correlation between the modified resources in a particular run. It is less natural than the following approach and could make querying more difficult (feedback on querying, please).
```

Comment:
If I read correctly, 30 files can be edited by the end-users. If values are concatenated into a single `asset` label, we can have many more values than 30 which would blow up cardinality.